src/category_theory/monoidal/subcategory.lean

```diff
@@ -0,0 +1,159 @@
+/-
+Copyright (c) 2022 Antoine Labelle. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Labelle
+-/
+import category_theory.monoidal.braided
+import category_theory.concrete_category.basic
+
+/-!
+# Full monoidal subcategories
+
+Given a monidal category `C` and a monoidal predicate on `C`, that is a function `P : C → Prop`
+closed under `𝟙_` and `⊗`, we can put a monoidal structure on `{X : C // P X}` (the category
+structure is defined in `category_theory.full_subcategory`).
+
+When `C` is also braided/symmetric, the full monoidal subcategory also inherits the
+braided/symmetric structure.
+-/
+
+universes u v
+
+namespace category_theory
+
+namespace monoidal_category
+
+open iso
+
+variables (C : Type u) [category.{v} C] [monoidal_category C]
+
+/--
+A property of objects of a monoidal category `C` which is closed under `𝟙_` and `⊗`.
+-/
+@[nolint has_inhabited_instance]
```

Comment:
I agree `has_inhabited_instance` is annoying, but it does serve some purpose. How about you provide an inhabited instance with the trivial predicate?